src/ui/components/task.svelte

```diff
@@ -17,7 +17,7 @@
 </script>
 
 <div
-  style:height="{$height}px"
+  style:height="{($height + $offset >= 1080) ? 1080-$offset : $height}px"
```

Comment:
1. Please hide all the height calculations in the `useTask` hook to keep the view separate from the reactive logic.
2. This won't work for different zoom levels & different start hours. Your calculation should include that into account.
3. Your code should be formatted with Prettier. Run `npm run format` before committing.